File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/interpret/TerminalExpression.java
package jv.com.tencent.cain.design_pattern.interpret;

/**
 * @author cainjiang
 * @date 2020/8/19
 */
public class TerminalExpression extends AbstractExpression {
    @Override
    void interpret(Context context) {
        System.out.println("终端解释器");
        System.out.println(context.toString());
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/interpret/TerminalExpression.kt
package kt.com.tencent.cain.desigin_pattern.interpret

/**
 * @author : jiangyu
 * @date   : 2020/8/19
 * @desc   : xxx
 */
class TerminalExpression : AbstractExpression() {
    override fun interpret(ctx: Context) {
        println("终端解释器, ctx：$ctx")
    }
}